interview-questions.md

```diff
@@ -8,44 +8,44 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 2. What is a gem?
 
-Your answer:
+Your answer:  A gem is something that we install onto ruby in order for it to have more uses than it comes with out of the box. An example of this would be in ruby testing, or rspec. This needs to be installed, but once you have it it is very useful. 
 
-Researched answer:
+Researched answer: A gem is  a package that ruby developers will download in order to add more onto what ruby comes standard with. RubyGems is the package manager that people use to do this. One example of this is rpsec, which allows people to test their code, similarly to how yarn works for javascript. Gems are generally used to modify or add onto what ruby, or ruby applications, can already do. 
 
 3. What is Ruby on Rails?
 
-Your answer:
+Your answer: Ruby on rails is a way of translating ruby code into a back end program so that we can use it to transfer data. It's kind of like react is to front end, ruby on rails is to back end. 
 
-Researched answer:
+Researched answer: Ruby on rails is an open source application that allows the use of Ruby to make full stack web applications. It employs the idea that while it may not be as coustomizable, it will provide a framework for everything you will need to do and thus make it a lot easier. It is similar to how many programmers use mac computers, because while something like a PC or linux computer would be far more customizable, that also opens it up to having problems in the future that will be harder to solve. Rails takes this same priciple, but applies it to the code itself. 
 
 4. What is a relational database? Are there other kinds of databases?
 
-Your answer:
+Your answer: A relational database is one that translates one language into something a database can read, then translates it back so that the programmer can read it. 
 
-Researched answer:
+Researched answer: A relational database is an idea that when data is stored, you can use relationships between two things to access it. An example of this is in a username that one might create on a website. The customer creates their username, and it is stored as a primary key, but while they do that a copy is made on the hard drive of the websites server somehwere, stored as a foreign key. Now that there is a relationship between these two things, they can have more data that is stored from that primary key and foreign key relationship. There are other kinds of databases, such as a non relational database. 
```

Comment:
This is very close! I really appreciate how well you articulated yourself beacuse it helps me out clearing some stuff up. 

In a table called User we would have:
- a primary key that is different for each user
- a username which would be a string
- a password which would be a string
- and maybe some specific other data

If I wanted my User to be able to make Post's, I would make another Table called Posts. 
I can associatie the tables using model relationships like the User has many posts and each post belongs to a User. 

My post table would have something like: 
- primary key for tracking all the different posts
- a title which is a string
- text which is a string or text
- and a foreign key for identifying who the author / User is that wrote it.